dotnet/src/SemanticKernel.Abstractions/Search/KernelSearchResults.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel.Search;
+
+/// <summary>
+/// Represents the search results returned from a <see cref="ITextSearch" /> service.
+/// </summary>
+[Experimental("SKEXP0001")]
+public class KernelSearchResults<T>
+{
+    /// <summary>
+    /// The total count of results found by the search operation, or null
+    /// if the count was not requested.
+    /// </summary>
+    public long? TotalCount { get; internal set; }
+
+    /// <summary>
+    /// The inner content representation. Use this to bypass the current abstraction.
+    /// </summary>
+    /// <remarks>
+    /// The usage of this property is considered "unsafe". Use it only if strictly necessary.
+    /// </remarks>
+    [JsonIgnore]
+    public object? InnerContent { get; }
```

Comment:
Considering that results is in an IAsyncEnumerable, having this just as a regular object may force an ineffeciency on us.  E.g. Let's say we're streaming the results using IAsyncEnumerable, but in order to populate InnerContent we have to wait until the data is fully streamed before we can get the full original dataset and set it here as an object.

Reply:
I'm going to leave out the breaking glass support for now as it's not clear we need this.